universal-application-tool-0.0.1/app/services/question/QuestionServiceImpl.java

```diff
@@ -64,7 +64,17 @@ public boolean addTranslation(
     if (!errors.isEmpty()) {
       return ErrorAnd.error(errors);
     }
-    question = questionRepository.updateQuestionSync(new Question(definition));
+    switch (question.getLifecycleStage()) {
```

Comment:
Suggestion (blocking): the purpose of this switch is a little complicated, please add a comment for it